symbolic-symcache/src/writer.rs

```diff
@@ -205,10 +206,15 @@ where
             .debug_session()
             .map_err(|e| SymCacheError::new(SymCacheErrorKind::BadDebugFile, e))?;
 
+        let mut seen_ranges = BTreeSet::new();
```

Comment:
Would it be enough to just track the function start address, provided that we just deal with top-level functions?